schema/spatem/spatem_schema_2-0-0.json

```diff
@@ -0,0 +1,327 @@
+              "states": {
+                "type": "array",
+                "description": "Each Movement is given in turn and contains its signal phase state, mapping to the lanes it applies to, and point in time it will end, and it may contain both active and future states.",
+                "minItems": 1,
+                "maxItems": 255,
+                "items": {
+                  "type": "object",
+                  "description": "MovementState is used to convey various information about the current or future movement state of a designated collection of one or more lanes of a common type. This is referred to as the GroupID. Note that lane object, types supported include both motorized vehicle lanes as well as pedestrian lanes and dedicated rail and transit lanes. Of the reported data elements, the time to change (the time remaining in the current state) is often of the most value. Lanes with a common state (typically adjacent sets of lanes in an approach) in a signalized intersection will have individual lane values such as total vehicle counts, summed. It is used in the SPAT message to convey every active movement in a given intersection so that vehicles, when combined with certain map information, can determine the state of the signal phases.",
+                  "required": [
+                    "signal_group",
+                    "state_time_speed"
+                  ],
+                  "properties": {
+                    "movement_name": {
+                      "type": "string",
+                      "description": "Human readable name for this movement. For debug use only."
+                    },
+                    "signal_group": {
+                      "description": "Used to map to lists of lanes (and their descriptions) which this MovementState data applies to.",
+                      "$ref": "../dsrc/dsrc_schema_2-0-0.json#/definitions/signal_group_id"
+                    },
+                    "state_time_speed": {
+                      "type": "array",
+                      "description": "Consisting of sets of movement data with SignalPhaseState, TimeChangeDetail and AdvisorySpeed Note: one or more of the movement events may be for a future time and that this allows conveying multiple predictive phase and movement timing for various uses for the current signal group.",
+                      "minItems": 1,
+                      "maxItems": 16,
+                      "items": {
+                        "type": "object",
+                        "description": "MovementEvent contains details about a single movement. It is used by the movement state to convey one of number of movements (typically occurring over a sequence of times) for a SignalGroupID.",
+                        "required": [
+                          "event_state"
+                        ],
+                        "properties": {
+                          "event_state": {
+                            "type": "integer",
```

Comment:
The DSRC defines this as an integer, indeed, but the defintion is that of an enumerated type. In this case, maybe it makes more sense for us to use a string with en umerated values, like so:
```json
"event_state": {
  "type": string",
  "enum": [
    "unavailable",
    "dark",
    "stop-Then-Proceed",
    ...
  ]
}
```
Thoughts? If so, then do we want it in a separate commit, like we do for the bit-fields-turned-lists-of-string?

Reply:
Hmm.. If we do so, there are so many other places where we'd need to change it too... Maybe that was not a great idea of me...

Reply:
> maybe we should discuss the implications for our other JSON messages first and leave it for a next version?

Yes, keep it as-is for now.